### DISCUSSION TITLE
How to mock keycloak login?

### DISCUSSION THREAD

**Discussion by etosdan** [mswjs/msw#1731] [Q&A] [answered]
How do I mock keycloak login?

**Comment by kettanaito** [comment, marked as answer]
Hi, @etosdan. 

MSW is designed to intercept HTTP requests. If you use the [REST API for Keycloak](https://developers.redhat.com/blog/2020/11/24/authentication-and-authorization-using-the-keycloak-rest-api), you will be able to intercept any HTTP requests your application makes and mock their responses to be what Keycloak would send in case of a successful/unsuccessful authorization. Look up how to use that service via a REST API, and then follow the [Mocking REST API](https://mswjs.io/docs/getting-started/mocks/rest-api) step of our tutorial to translate that API into request handlers.
